src/main/java/org/springframework/integration/dsl/LambdaMessageProcessor.java

```diff
@@ -125,6 +126,9 @@ public Object processMessage(Message<?> message) {
 		try {
 			return this.method.invoke(this.target, args);
 		}
+		catch(InvocationTargetException e) {
+			throw new MessageHandlingException(message, e.getCause());
+		}
```

Comment:
Nothing to complain about!
Merging.
I wonder if we have similar issue in other reflection method invocation cases...